eth/stagedsync/stage_senders.go

```diff
@@ -383,16 +383,7 @@ func PruneSendersStage(s *PruneState, tx kv.RwTx, cfg SendersCfg, ctx context.Co
 		defer tx.Rollback()
 	}
 
-	sn := cfg.blockRetire.Snapshots()
-	// With snapsync - can prune old data only after snapshot for this data created: CanDeleteTo()
-	if sn != nil && sn.Cfg().Enabled && sn.Cfg().Produce {
-		if err := cfg.blockRetire.PruneAncientBlocks(tx); err != nil {
-			return err
-		}
-		if err := retireBlocksInSingleBackgroundThread(s, cfg, ctx, tx); err != nil {
-			return fmt.Errorf("retireBlocksInSingleBackgroundThread: %w", err)
-		}
-	} else if cfg.prune.TxIndex.Enabled() {
+	if cfg.prune.TxIndex.Enabled() {
```

Comment:
without `if sn.Cfg().Enabled` part - senders will be pruned before retire:
- senders.prune
- creation of new snapshot starting: and it will recover pruned senders (it's slow and cpu intensive)

Reply:
I thought by adding snapshots prune stage before the sender prune stage would avoid this issue. Since we would first retire blocks and then prune all the information.